src/sentry/data_export/tasks.py

```diff
@@ -21,157 +34,207 @@
 logger = logging.getLogger(__name__)
 
 
-@instrumented_task(name="sentry.data_export.tasks.assemble_download", queue="data_export")
+@instrumented_task(
+    name="sentry.data_export.tasks.assemble_download",
+    queue="data_export",
+    default_retry_delay=30,
+    max_retries=3,
+)
 def assemble_download(
     data_export_id,
     export_limit=1000000,
     batch_size=SNUBA_MAX_RESULTS,
+    offset=0,
+    bytes_written=0,
     environment_id=None,
     **kwargs
 ):
-    # Get the ExportedData object
     try:
-        logger.info("dataexport.start", extra={"data_export_id": data_export_id})
-        metrics.incr("dataexport.start", tags={"success": True}, sample_rate=1.0)
+        if offset == 0:
```

Comment:
Nit: We check `offset == 0` a lot... maybe just store it as `first_page = offset == 0` and reference that instead.